hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/StatsFileSizeProcedure.scala

```diff
@@ -54,8 +55,22 @@ class StatsFileSizeProcedure extends BaseProcedure with ProcedureBuilder {
     val globRegex = getArgValueOrDefault(args, parameters(1)).get.asInstanceOf[String]
     val limit: Int = getArgValueOrDefault(args, parameters(2)).get.asInstanceOf[Int]
     val basePath = getBasePath(table)
-    val fs = HoodieTableMetaClient.builder.setConf(jsc.hadoopConfiguration()).setBasePath(basePath).build.getFs
-    val globPath = String.format("%s/%s/*", basePath, globRegex)
+    val metaClient = HoodieTableMetaClient.builder.setConf(jsc.hadoopConfiguration()).setBasePath(basePath).build
+    val fs = metaClient.getFs
+    val isTablePartitioned = metaClient.getTableConfig.isTablePartitioned
+    val maximumPartitionDepth = if (isTablePartitioned) metaClient.getTableConfig.getPartitionFields.get.length else 0
+    val globPath = (metaClient.getTableConfig.isTablePartitioned, globRegex) match {
```

Comment:
```suggestion
    val globPath = (isTablePartitioned, globRegex) match {
```

Reply:
Done!

Reply:
@danny0405 I've addressed this comment, can you please help to take a look at this again? Thank you!